### PR TITLE
feat: support multiple classes with annotations

### LINKED ISSUE #83: feat: support multiple classes extending `CLightningPlugin`
Currently, and if I am not mistaken, the library only supports a single `Plugin` class on the whole classpath.
In some cases it is desirable to have multiple classes extending `CLightningPlugin`, even if only one is  actually active (by invoking `start()`).

The reasons seems to be that `CLightningPlugin#registerMethod` collects all classes containing any of the annotations (`Subscription`, `Hook`, etc.) and subsequently tries to invoke it on the currently active plugin object, even if it might not have the method defined itself.

I assume the plugin still works and will just report false or duplicate information in the `getmanifest` response.
Would it be desirable to report the correct information and support multiple classes?

### PR DESCRIPTION
**Pull request by theborakompanioni**
Resolves #83.

Before this commit, the lib scanned all classes and registered any annotated method (`RPCMethod`, `Subscription`, `Hook`, etc.) found. This would lead to errors when multiple classes are on the classpath containing the annotations and were invoked on them.
When this commit is applied, the lib will only register methods with annotations that are contained in the class on which `start` has been invoked.

This should not be a breaking change as the library only supports one instance of a `Plugin` class at a time, but makes in possible to "at least" contain multiple classes and only creating an instance of one of them.

### CODE REVIEW

File: src/main/java/jrpc/clightning/plugins/CLightningPlugin.java [hunk 1]
```diff
@@ -158,15 +159,22 @@ public void start() {
...
+                      it -> {
+                        Subscription subscription = it.getAnnotation(Subscription.class);
+                        String event = subscription.notification();
+                        String methodName = notificationRequest.get("method").getAsString();
+                        return methodName.equals(event);
+                      })
+                  .collect(Collectors.toList());
+
+          for (Method method : eligibleMethods) {
+            method.invoke(this, notificationRequest);
```
Comment (vincenzopalazzo) [inline src/main/java/jrpc/clightning/plugins/CLightningPlugin.java:177]: Now that we are rewriting this code maybe we can fix my previous TODO by moving this search code inside a function as I did for other annotation?

**Review by vincenzopalazzo — COMMENTED** [review]
I like update my old wat to write java code :)

Thanks

File: src/main/java/jrpc/clightning/plugins/CLightningPlugin.java (hunk 1 above)
Comment (theborakompanioni) [reply src/main/java/jrpc/clightning/plugins/CLightningPlugin.java:177]: If that is okay for you, I'll do that in a follow-up PR. :pray:

Comment (vincenzopalazzo) [reply src/main/java/jrpc/clightning/plugins/CLightningPlugin.java:177]: it is perfect :)

**Review by vincenzopalazzo — APPROVED** [review]
Looks good to me, and sorry for delay but I did not receive the update from Github that you was ready for review.

So this is good to me, lets wait the integration testing finish and I will merge it

**Pull request merged**